### PR TITLE
Treat control characters as width 1, fixes #16

### LINKED ISSUE #16: unicode control characters trigger `debug_assert_eq`
Hello,

Thanks for writing the truncation code ^^

During testing of this library, I noticed that carefully crafted inputs containing control caracters can trigger an assertion. This only happens in in debug mode => not a runtime panic for release. The used character is [`End of Medium`](https://www.compart.com/en/unicode/U+0019).

The assertion is located here
https://github.com/Aetf/unicode-truncate/blob/9e49ef4f9ac4e140b854a655daf4bc1d096adf5d/src/lib.rs#L174

A minimal (failing) testcase is the following

```rust
#[cfg(test)]
mod tests {
  #[test]
  fn zero_width_control_char() {
      std::assert_eq!("\u{0019}".unicode_truncate(2), ("\u{0019}", 0));
  }
}
```

This happens because at said point (expanded for better readability)

- ```
  let (byte_index, new_width) = (1, 0);
  ```
- ```
  let result = "\u{0019}".get(..1).unwrap();
  ```
- ```
  let result = "\u{0019}";
  ```
- => thus the `result.width()` (=1) is larger than the `new_width` (=0).

This case happens as `(self.len(), 0)` is appended to the interator.
This makes the iterator (as intended by the doccomment) eager for trailing zero width characters.
=> The asserted invariant does not hold anymore.

A possible fix would be to change the assertion to
```diff
- debug_assert_eq!(result.width(), new_width);
+ debug_assert!(result.width() >= new_width);
```

I can provide a PR with this fix if you'd prefer this

### PR DESCRIPTION
**Pull request by Aetf**
This is consistent with how unicode-width handles string width vs char width.

See also https://github.com/unicode-rs/unicode-width/pull/45

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -44,3 +44,5 @@ harness = false
 codegen-units = 1
 lto = true
 
+[profile.test]
+debug-assertions = true
```
Comment (EdJoPaTo) [inline Cargo.toml:48]: I am curious why this change was done. Test is by default dev and that's with debug assertions? https://doc.rust-lang.org/cargo/reference/profiles.html#test

Comment (Aetf) [reply Cargo.toml:48]: Oh, I forgot to remove this. I was debugging why the assertion didn't trigger when I tested locally. It turned out that the unicode-width in my local lock file was still the old version.
